src/sys/socket/sockopt.rs

```diff
@@ -368,7 +368,7 @@ impl<'a> Set<'a, usize> for SetUsize {
     }
 
     unsafe fn ffi_ptr(&self) -> *const c_void {
```

Comment:
Oh, sorry, I misread. You suggest the traits themselves be unsafe? The docs say:

> If unsafe code cannot reasonably expect to defend against a bad implementation of the trait, then marking the trait unsafe is a reasonable choice.

Given this and other topics around unsafe, it does appear that the traits should be unsafe and all the functions should be safe.